sql/auth/sql_user_table.cc

```diff
@@ -2011,9 +2022,10 @@ int open_grant_tables(THD *thd, TABLE_LIST *tables,
 
     for (auto i = 0; i < ACL_TABLES::LAST_ENTRY; i++)
       tables[i].updating = false;
+
+    thd->mdl_context.release_transactional_locks();
```

Comment:
Can we have this logic of temporary release of lock and re-acquire to apply only for WSREP enabled threads?
```suggestion
    if (WSREP(thd)) thd->mdl_context.release_transactional_locks();
```

Reply:
Please imagine the flow for non-slave thread. 
This part is executed only for slave thread.

Reply:
I meant if we can skip this temporary release of locks if it is not a WSREP enabled thread, but yet an applier thread i(n other words async replica threads).

Reply:
Currently it looks that **lock & check-unlock & reacquire** happens even for non-wsrep async replica threads
``` c++
if (slave_thread) {
  if (acl_tables_setup_for_write_and_acquire_mdl(thd, tables)) return -1;
  ..
  thd->mdl_context.release_transactional_locks();
}

if (WSREP(thd) && wsrep_to_isolation_begin()) {..}
...

if (acl_tables_setup_for_write_and_acquire_mdl(thd, tables)) return -1;
```